### PR TITLE
move running tests to a separate jenkins job

### PR DESCRIPTION
**Pull request by jakubgs**
We'll run them in parallel, should make things faster and allow us to drop Travis.

Status: WIP

### CODE REVIEW

**Review by mandrigin — APPROVED** [review]

**Comment by mandrigin** [comment]
@jakubgs is it still WIP?

**Comment by jakubgs** [comment]
@mandrigin I talked to @adambabik and he thinks that what we should do is run it all as one job. As in, first run the tests, then the builds. But that kinds breaks our workflow.

1. That means all builds will show up on github as one build
2. It increases the time to build artifacts by ~12 minutes
3. You would have to re-run the whole thing if one of the platform fails
4. It complicates the flow in Jenkinsfiles

I think it's better to run tests as a separate job. If we set the requirements for the PRs right we won't be able to merge without them passing.

**Comment by mandrigin** [comment]
@jakubgs yeah, I tend to agree with you, because what you want to be able to do is to restart all the jobs separately. Some teams go as far as having unit-tests, lint, e2e as separate jobs too.
We have codeclimate as a separate job and the packages check. I don't think with status-go it will add any overhead to our CI.

**Comment by mandrigin** [comment]
@adambabik what do you think?

File: _assets/ci/Jenkinsfile.tests [hunk 1]
```diff
@@ -0,0 +1,69 @@
+pipeline {
```
Comment (adambabik) [inline _assets/ci/Jenkinsfile.tests:1]: Doesn't it need to be used somewhere?

**Review by adambabik — APPROVED** [review]

File: _assets/ci/Jenkinsfile.tests (hunk 1 above)
Comment (jakubgs) [reply _assets/ci/Jenkinsfile.tests:1]: What do you mean?

Comment (adambabik) [reply _assets/ci/Jenkinsfile.tests:1]: I am just wondering where and how this Jenkinsfile is used.

Comment (jakubgs) [reply _assets/ci/Jenkinsfile.tests:1]: https://ci.status.im/job/status-go/job/prs/job/tests/view/change-requests/ - here

Comment (jakubgs) [reply _assets/ci/Jenkinsfile.tests:1]: And you can see it below, as `jenkins/prs/tests`.

Comment (adambabik) [reply _assets/ci/Jenkinsfile.tests:1]: Oh ok, configured directly in Jenkins. All clear.

**Pull request merged**